config_builder.py

```diff
@@ -0,0 +1,251 @@
+#!/opt/datadog-agent/embedded/bin/python
+'''
+This script is used to generate the configuration of the datadog agent, its
+integrations and other moving parts.
+'''
+
+from os import getenv, environ
+from os.path import isfile
+import logging
+from urllib2 import urlopen, URLError, HTTPError
+from socket import getdefaulttimeout, setdefaulttimeout
+from ConfigParser import ConfigParser
+import yaml
+
+class ConfBuilder(object):
+    '''
+    This class manages the configuration files
+    '''
+    def __init__(self):
+        # excludes from the generic variables parsing the ones that have a
+        # certain logic warpped around them
+        self.exclude_from_generic = [
+            'DD_API_KEY', 'DD_API_KEY_FILE', 'DD_HOME',
+            'DD_START_AGENT', 'DD_LOGS_STDOUT'
+            ]
+        dd_agent_root = '/etc/dd-agent'
+        dd_home = getenv('DD_HOME')
+        if dd_home is not None:
+            dd_agent_root = '{}/agent'.format(dd_home)
+        self.datadog_conf_file = '{}/datadog.conf'.format(dd_agent_root)
+        self.supervisor_conf_file = '{}/supervisor.conf'.format(dd_agent_root)
+        self.conf_d = '{}/conf.d'.format(dd_agent_root)
+        # This will store the config parser object that is used in the different functions
+        self.config = None
+
+    def set_instance_property(self, integration, property_key, property_value):
```

Comment:
Not sure what's hacky in that. It just opens the integration config file (from the example or from the existing one if any) and updates the property for the instance (or the several instances if several ones are already defined).
It looks more complex but in the long term it would be more maintainable than having to change 6 sed in 6 entrypoint.sh no?
That function is generic and is only used in 'build_integration_conf` which is just a rewrite in python of their shell counterpart present in several of the entrypoints. Looks more maintainable on the long term no?

And when we switch to allow environment variables (which we do for some in `build_integration_conf`) we wouldn't really have to modify it.

Note: the generic `DD_<integration_name>_<property_name>` idea have not been implemented this PR.